app/controllers/spree/admin/return_authorizations_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module Admin
+    class ReturnAuthorizationsController < ResourceController
+      belongs_to 'spree/order', find_by: :number
+
+      update.after :associate_inventory_units
+      create.after :associate_inventory_units
+
+      def fire
+        @return_authorization.public_send("#{params[:e]}!")
```

Comment:
This is a terrible pattern for security. I can't even see any authorisation here. The client can call any method ending in `!` on this object. Good to have that visible in our code now.

Reply:
Um... I'm not sure we should merge this as is?

Reply:
Or at least make an issue to change this line, and put it at the top of Tech Debt.

Reply:
this is live code .
this is not only spree 2 code but also spree master code 🙈  🤕 
https://github.com/spree/spree/blob/98fa319554e9bad01fee9a9f56500eea99847d29/backend/app/controllers/spree/admin/return_authorizations_controller.rb#L11

the security risk is really just adding return authorizations.

I'll create a tech debt issue.
https://github.com/openfoodfoundation/ofn-security/issues/21